spec/models/miq_provision_workflow_spec.rb

```diff
@@ -53,20 +61,77 @@
         it "should encrypt fields" do
           password_input = "secret"
           request = ManageIQ::Providers::Vmware::InfraManager::ProvisionWorkflow.from_ws(
-            "1.1", admin, "name=template", "vm_name=spec_test|root_password=#{password_input}",
-            "owner_email=admin|owner_first_name=test|owner_last_name=test", nil, nil, nil, nil)
+            "1.1", admin, {'name' => 'template'}, {'vm_name' => 'spec_test', 'root_password' => "#{password_input}"},
+            {'owner_email' => 'admin'}, {'owner_first_name' => 'test'},
+            {'owner_last_name' => 'test'}, nil, nil, nil, nil)
 
           MiqPassword.encrypted?(request.options[:root_password]).should be_true
           MiqPassword.decrypt(request.options[:root_password]).should == password_input
         end
 
+        it "should set values when extra '|' are passed in" do
+          request = ManageIQ::Providers::Vmware::InfraManager::ProvisionWorkflow.from_ws(
+            "1.1", admin, {'name' => 'template'}, {'vm_name' => 'spec_test'},
+            nil, nil, {'abc' => 'tr|ue', 'blah' => 'nah'}, nil, nil)
+
+          expect(request.options[:ws_values]).to include(:abc => "tr|ue")
+        end
+
+        it "should set values when extra '|' are passed in for multiple values" do
+          request = ManageIQ::Providers::Vmware::InfraManager::ProvisionWorkflow.from_ws(
+            "1.1", admin, {'name' => 'template'}, {'vm_name' => 'spec_test'},
+            nil, nil, {'abc' => 'tr|ue', 'blah' => 'na|h'}, nil, nil)
+
+          expect(request.options[:ws_values]).to include(:blah => "na|h")
+        end
+
         it "should set values" do
           request = ManageIQ::Providers::Vmware::InfraManager::ProvisionWorkflow.from_ws(
-            "1.1", admin, "name=template", "vm_name=spec_test",
-            nil, nil, "abc=true", nil, nil)
+            "1.1", admin, {'name' => 'template'}, {'vm_name' => 'spec_test'},
+            nil, nil, {'abc' => 'true', 'blah' => 'nah'}, nil, nil)
 
           expect(request.options[:ws_values]).to include(:abc => "true")
         end
+
+        it "should set values correctly when a string is passed" do
```

Comment:
Same for this test and the next.  The string parsing is checked in the `when all args are passed in as a string` test and this is not buying anything extra.  And the use-case is passing all strings (old-style) or all hashes (new-style).